### DISCUSSION TITLE
Width Resize ?

### DISCUSSION THREAD

**Discussion by chazapp** [utk-se/CodeRibbon#83] [Q&A] [answered]
Hi, i'm trying out CodeRibbon and love the idea so far, but why am I unable to resize the width of a pane ? I'm not an Atom user but i'm trying to start, I've been looking everywhere in Atom's docs, but I can't resize a code pane's width like i would do the Project's pane. Any idea why, and how to fix this ?

**Comment by AZHenley** [comment, marked as answer]
It is by design. One of the rationales of CodeRibbon is to reduce the time spent doing window/tab management activities, such as the "jigsaw" problem with resizable panes. 

You can read more here: [Patchworks CHI'14](https://web.eecs.utk.edu/~azh/pubs/Henley2014CHI_Patchworks.pdf).

This is a tricky balancing act of freedom and constraint, so we are always open to suggestions though!

**Reply by chazapp** [reply]
This constraint is something I cannot accept from my IDE. If I have a Dockerfile with often <50 chars lines open along side TS project files with >120 chars lines, half of my screen is kept empty for seemingly no reason at all. I'll keep checking this very promising project in the future for updates ; I would not be surprised to hear others complain about this.

**Reply by robobenklein** [reply]
That's totally expected for some people, this style of editor I would generally advertise to developers who work with a lot of files that all generally respect some line length limit, i.e. for python we generally see lines ~80 characters, so CodeRibbon automatically displays as many patches as it can while keeping at least 80 characters width in each.

If you have a screen showing 3 patches horizontally you could change either CodeRibbon's settings or Atom's preferred-line-length to show only two patches horizontally.
